tests/expected/attr-invariant/check_arbitrary/check_arbitrary.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Check that the `kani::invariant` attribute automatically generates the
+//! `Arbitrary` implementation, and also that the values it generates respect
+//! the type invariant.
+
+extern crate kani;
+use kani::Invariant;
+
+#[kani::invariant(*x >= 0 && *y >= 0)]
```

Comment:
This example is repeated too many times in the PRs :) Can we add examples with more interesting cases, e.g. a predicate that refers to multiple fields (e.g. `*x == *y`), or a struct with types other than `i32` (e.g. `char`) or user-defined types?

Reply:
Added a few more tests which do more interesting things.